snmp/datadog_checks/snmp/snmp.py

```diff
@@ -118,15 +119,27 @@ def _get_profiles_mapping(self):
                     profiles_by_oid[sys_object_oid] = name
         return profiles_by_oid
 
+    @property
+    def _mib_loader(self):
+        # type: () -> MIBLoader
+        """
+        A lazily-loaded shared MIB loader instance.
+        """
+        if not hasattr(self, "_mib_loader_instance"):
+            self._mib_loader_instance = MIBLoader()
+        return self._mib_loader_instance
+
     def _build_config(self, instance):
         # type: (dict) -> InstanceConfig
+        loader = self._mib_loader if self.shared_mib_builder else MIBLoader()
```

Comment:
Unless I miss something, this would only share the loader when you use python autodiscovery, no?

Reply:
Ah, probably so. This means the shared instance should probably live as a class attribute on `MIBLoader`. Lemme fix this…